### ISSUE TITLE
camel-kamelet: improve global properties handling

### ISSUE THREAD

**Issue by lburgazzoli** [apache/camel-k-runtime#560]
The way the `kamelet` component handles properties as today is:

```
camel.kamelet.${templateId}
camel.kamelet.${templateId}.${routeId}
```

But there are some issue with that:

1. the `routeId` may be confusing as it can be used for setting a template property and as a root for the materialized route properties
2. it uses a non standard prefix as all the component use `camel.component.${scheme}`

A possible alternative is to to leverage camel's capabilities to use map, something like:

```
camel.component.kamelet.template-properties[${templateId}][foo] = bar
camel.component.kamelet.route-properties[${routeId}][foo] = bar
```

However it looks quite ugly, a better option would be:

```
camel.component.kamelet.template-properties[${templateId}].foo = bar
camel.component.kamelet.route-properties[${routeId}].foo = bar
```

But I don't think camel is currently able to handle this case so we should probably enhance the properties binding support, as example we can mark a component to be `PropertiesAware` so the binding process should not try to bind the entire object tree but it delegates the mapping to the receiver object.

```java
public interface PropertiesAware {
    void setProperties(Map<String, String> properties);
    Map<String, String> properties getProperties(Map<String, String> properties);
}
```

Assuming the component would have been something like:

```java
public class KameletComponent extends DefaultComponent {
    private Map<String, PropertiesHolder> templateProperties;
    private Map<String, PropertiesHolder> routeProperties;

    // getter/Setter omitted
}

public class PropertiesHolder implements PropertiesAware {
    @Override
    public void setProperties(Map<String, String> properties) { ... }
    @Override
    public Map<String, String> properties getProperties(Map<String, String> properties) { ... }
}
```

The binding process then can stop at `camel.component.kamelet.template-properties[${templateId}]` and `
camel.component.kamelet.route-properties[${routeId}]` and any further processing should be eventually done by the receiver.

@davsclaus does this make any sense ?
@nicolaferraro would the new format be problematic for the operator ? (we'll keep the old way for some time in any case)

**Comment by davsclaus** [comment]
On top of my head there is some components that have property prefix, like camel-http with http client options
https://github.com/apache/camel/blob/master/components/camel-http/src/main/java/org/apache/camel/component/http/HttpComponent.java#L406

Maybe we could have something like that, then its

```
camel.component.kamelet.template-properties.myTemplate.foo = bar
camel.component.kamelet.template-properties.myTemplate.code = 1234
camel.component.kamelet.route-properties.myRoute.foo = bar
```

Then `template-properties` and `route-properties` are just a Map which we then use that extract code (see link above) and in the kamelet component we can then do the binding we want.

**Comment by lburgazzoli** [comment]
oh, that's cool.  will test it

**Comment by lburgazzoli** [comment]
@davsclaus do you think we can have something for this in 3.7 ?

**Comment by davsclaus** [comment]
Yeah have you tried that extract property thingy?

**Comment by lburgazzoli** [comment]
not yet

**Comment by davsclaus** [comment]
Okay if you dont find the time and when I get the compiled simple done, then I can take a look

**Comment by lburgazzoli** [comment]
Ok, so after we get [CAMEL-15886](https://issues.apache.org/jira/browse/CAMEL-15886), [/CAMEL-15887](https://issues.apache.org/jira/browse/CAMEL-15887) solved and we get camel 3.7 out then we can get this working

**Comment by lburgazzoli** [comment]
Fixed by https://github.com/apache/camel-k-runtime/pull/573
